src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/GenerateComponentLocationAnalysisOperation.java

```diff
@@ -0,0 +1,49 @@
+package com.synopsys.integration.detect.workflow.componentlocationanalysis;
+
+import com.synopsys.integration.blackduck.api.generated.view.DeveloperScansScanView;
+import com.synopsys.integration.detect.configuration.DetectUserFriendlyException;
+import com.synopsys.integration.detect.configuration.enumeration.ExitCodeType;
+import com.synopsys.integration.detect.workflow.bdio.BdioResult;
+import com.synopsys.integration.detect.workflow.file.DetectFileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+public class GenerateComponentLocationAnalysisOperation {
+    private static final String CLL_INPUT_FILE_NAME = "components-source.json";
+    private static final String CLL_OUTPUT_FILE_NAME = "components-with-locations.json";
+    private final BdioToCLLComponentTransformer bdioTransformer;
+    private final ScanResultToCLLComponentTransformer scanResultTransformer;
+
+    public GenerateComponentLocationAnalysisOperation(BdioToCLLComponentTransformer bdioTransformer, ScanResultToCLLComponentTransformer scanResultTransformer) {
+        this.bdioTransformer = bdioTransformer;
+        this.scanResultTransformer = scanResultTransformer;
+    }
+
+    public static File forNonPersistentOnlinePkgMngrScan(List<DeveloperScansScanView> rapidFullResults, File scanOutputFolder) throws DetectUserFriendlyException {
```

Comment:
I'd suggest replacing the `PkgMngr` terms everywhere with full forms (longer method names should be fine).
Another option is to name them as `Detector` as the `"DETECTOR"` value for `detect.tools` property also means a package manager scan.